src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -110,6 +114,38 @@ QString QgsOgrLayerItem::layerName() const
     return info.completeBaseName();
 }
 
+#ifdef HAVE_GUI
+QList<QAction *> QgsOgrLayerItem::actions()
+{
+  QList<QAction *> lst;
+  QAction *actionDeleteLayer = new QAction( tr( "Delete %1" ).arg( mName ), this );
+  connect( actionDeleteLayer, &QAction::triggered, this, &QgsOgrLayerItem::deleteLayer );
+  lst.append( actionDeleteLayer );
+  return lst;
+}
+
+void QgsOgrLayerItem::deleteLayer()
+{
+  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
+                              QObject::tr( "Are you sure you want to delete %1?" ).arg( mName ),
```

Comment:
What do you suggest? 
BTW a different wording can be tricky since this is abstracted in OGR: do you know a way to know if the GDAL/OGR layer we are deleting is a file or a table from a DB?